In [3]:
# BiocManager::install("tidyverse")

'getOption("repos")' replaces Bioconductor standard repositories, see
'?repositories' for details

replacement repositories:
    CRAN: https://cran.r-project.org


Bioconductor version 3.14 (BiocManager 1.30.18), R 4.1.3 (2022-03-10)

Installing package(s) 'tidyverse'

also installing the dependencies ‘rematch’, ‘gargle’, ‘cellranger’, ‘ids’, ‘vroom’, ‘tzdb’, ‘selectr’, ‘broom’, ‘dtplyr’, ‘forcats’, ‘googledrive’, ‘googlesheets4’, ‘haven’, ‘modelr’, ‘readr’, ‘readxl’, ‘reprex’, ‘rvest’


Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done

Old packages: 'FNN', 'gtools', 'randomForest'



In [1]:
# library(SingleCellExperiment)
# library(devtools)
library(TED)

Loading required package: snowfall

Loading required package: snow

Warning message:
“replacing previous import ‘gplots::lowess’ by ‘stats::lowess’ when loading ‘TED’”


In [21]:
#load in BP demo data in vignette
load("/home/cke/BPexample_gbm.rdata")

In [33]:
length(t(ref.dat))

In [19]:
?run.Ted
# ?cleanup.genes

In [3]:
df_Puram<-read.table("/home/cke/Puram/HNSCC2PuramGSE103322_HNSCC_exp.tsv",sep='\t',header=TRUE)

In [4]:
df_Puram

X               HN28_P15_D06_S330_comb HN28_P6_G05_S173_comb
1     ENSG00000121410 0.000000               0.00000              
2     ENSG00000268895 0.000000               0.00000              
3     ENSG00000148584 0.000000               0.00000              
4     ENSG00000175899 9.683000               7.58300              
5     ENSG00000245105 0.176320               0.00000              
6     ENSG00000166535 0.000000               0.00000              
7     ENSG00000256069 0.000000               0.00000              
8     ENSG00000128274 2.248800               0.00000              
9     ENSG00000118017 0.000000               0.00000              
10    ENSG00000265544 0.000000               0.00000              
11    ENSG00000094914 4.542300               4.69200              
12    ENSG00000081760 3.564100               0.00000              
13    ENSG00000250420 0.000000               0.00000              
14    ENSG00000114771 0.000000               0.00000              
15    ENSG00000197953 0.000000               0.00000              
16    ENSG00000188984 0.000000               0.00000              
17    ENSG00000204518 0.000000               0.00000              
18    ENSG00000109576 0.000000               0.00000              
19    ENSG00000158122 0.000000               0.00000              
20    ENSG00000103591 0.000000               0.91533              
21    ENSG00000115977 0.038436               0.13488              
22    ENSG00000087884 0.000000               0.00000              
23    ENSG00000127837 0.000000               0.00000              
24    ENSG00000129673 0.000000               0.00000              
25    ENSG00000131043 0.000000               0.00000              
26    ENSG00000205002 0.211640               0.00000              
27    ENSG00000090861 0.000000               0.00000              
28    ENSG00000124608 0.000000               0.18777              
29    ENSG00000266967 0.000000               0.00000              
30    ENSG00000157426 0.000000               0.00000              
⋮     ⋮               ⋮                      ⋮                    
21784 ENSG00000197062 2.362300               0.00000              
21785 ENSG00000140265 0.109690               0.00000              
21786 ENSG00000186814 0.000000               0.00000              
21787 ENSG00000235109 0.000000               0.00000              
21788 ENSG00000140987 0.000000               0.00000              
21789 ENSG00000180532 0.000000               0.00000              
21790 ENSG00000131848 0.000000               0.00000              
21791 ENSG00000197213 0.000000               0.00000              
21792 ENSG00000137185 0.000000               0.00000              
21793 ENSG00000168612 0.000000               0.58688              
21794 ENSG00000163012 0.000000               0.00000              
21795 ENSG00000132801 0.000000               0.00000              
21796 ENSG00000132003 0.000000               0.00000              
21797 ENSG00000162415 0.000000               0.00000              
21798 ENSG00000130449 0.069015               0.00000              
21799 ENSG00000214941 0.000000               0.00000              
21800 ENSG00000214655 2.788100               0.00000              
21801 ENSG00000272589 0.000000               0.00000              
21802 ENSG00000153975 0.000000               0.00000              
21803 ENSG00000086827 0.000000               0.00000              
21804 ENSG00000174442 0.000000               0.00000              
21805 ENSG00000122952 0.000000               0.00000              
21806 ENSG00000198205 0.000000               0.00000              
21807 ENSG00000198455 0.000000               0.00000              
21808 ENSG00000070476 0.121680               0.19535              
21809 ENSG00000203995 0.163500               0.27977              
21810 ENSG00000162378 0.090853               3.93480              
21811 ENSG00000159840 4.274800               0.00000 

In [5]:
df_TCGA <- read.table("/home/cke/Puram/TCGA-HNSC.htseq_counts_exp2.tsv",sep='\t',header=TRUE)

In [6]:
df_TCGA

Ensembl_ID      TCGA.BB.4224.01A TCGA.H7.7774.01A TCGA.CV.6943.01A
1     ENSG00000000003 2237              2740             2686           
2     ENSG00000000005    2                 0                0           
3     ENSG00000000419 1606              1691             1649           
4     ENSG00000000457 1063               803              917           
5     ENSG00000000460 1208               317              402           
6     ENSG00000000938  256               330             1535           
7     ENSG00000000971  177              5472            15358           
8     ENSG00000001036 2371              2316             3490           
9     ENSG00000001084 2741              8370             4671           
10    ENSG00000001167 2515               816             2080           
11    ENSG00000001460 1273               455              686           
12    ENSG00000001461 1036              2173             2877           
13    ENSG00000001497 4205              2281             3950           
14    ENSG00000001561   36                71              352           
15    ENSG00000001617 5270             14807            10608           
16    ENSG00000001626   22                51                2           
17    ENSG00000001629 2033              2394             3963           
18    ENSG00000001630   30               533              340           
19    ENSG00000001631 1474               928             1249           
20    ENSG00000002016  956               144              276           
21    ENSG00000002079   71                 5               15           
22    ENSG00000002330 2176              1469             1792           
23    ENSG00000002549 1761             11512            23733           
24    ENSG00000002586 6575              6882            18744           
25    ENSG00000002587  364               308              963           
26    ENSG00000002726  134               308              547           
27    ENSG00000002745  194                73               19           
28    ENSG00000002746  116               203              226           
29    ENSG00000002822 1604              1181             3040           
30    ENSG00000002834 8014              8810            25075           
⋮     ⋮               ⋮                ⋮                ⋮               
60454 ENSGR0000223274 0                0                0               
60455 ENSGR0000223484 0                0                0               
60456 ENSGR0000223511 0                0                0               
60457 ENSGR0000223571 0                0                0               
60458 ENSGR0000223773 0                0                0               
60459 ENSGR0000225661 0                0                0               
60460 ENSGR0000226179 0                0                0               
60461 ENSGR0000227159 0                0                0               
60462 ENSGR0000228410 0                0                0               
60463 ENSGR0000228572 0                0                0               
60464 ENSGR0000229232 0                0                0               
60465 ENSGR0000230542 0                0                0               
60466 ENSGR0000234622 0                0                0               
60467 ENSGR0000234958 0                0                0               
60468 ENSGR0000236017 0                0                0               
60469 ENSGR0000236871 0                0                0               
60470 ENSGR0000237040 0                0                0               
60471 ENSGR0000237531 0                0                0               
60472 ENSGR0000237801 0                0                0               
60473 ENSGR0000263835 0                0                0               
60474 ENSGR0000263980 0                0                0               
60475 ENSGR0000264510 0                0                0               
60476 ENSGR0000264819 0                0                0       

In [7]:
df_TCGA_t<-t(df_TCGA)
colnames(df_TCGA_t)<-df_TCGA_t[1,]
df_TCGA_t <- df_TCGA_t[-1,]

In [8]:
df_TCGA_t

ENSG00000000003 ENSG00000000005 ENSG00000000419
TCGA.BB.4224.01A    2237               2            1606        
TCGA.H7.7774.01A    2740               0            1691        
TCGA.CV.6943.01A    2686               0            1649        
TCGA.CN.5374.01A   2086               1            2333         
TCGA.CQ.6227.01A   10167               9            3021        
TCGA.CV.6959.01A    1154               6            2766        
TCGA.F7.A61V.01A   1978               1            1762         
TCGA.CV.7413.01A    1930               1            1668        
TCGA.CV.7247.01A    1066               1            1760        
TCGA.CR.5249.01A    2538               0            1268        
TCGA.CQ.5331.01A    1465               0            2612        
TCGA.BB.8601.01A   1630               1            1227         
TCGA.CV.A45W.01A    2215               0            1139        
TCGA.CQ.7063.01A     724               0            1220        
TCGA.CN.5373.01A     718               0            1162        
TCGA.CV.A45T.01A    2120               3            2252        
TCGA.HD.A4C1.01A   1273               2            1793         
TCGA.CV.7438.11A    4370               3            2064        
TCGA.CV.5444.01A   1011               0            1488         
TCGA.BA.5556.01A    1646               1            2146        
TCGA.P3.A6SX.01A    1421               0            2413        
TCGA.CV.7242.11A    9952               9            2403        
TCGA.DQ.7592.01A    2923               0            2507        
TCGA.CV.6937.01A   1232               2            1212         
TCGA.QK.A8ZB.01A    2332               0            3280        
TCGA.CV.7432.01A    2032               0            2069        
TCGA.CV.7089.01A    3005               0            2056        
TCGA.CV.6939.11A    1612              53             729        
TCGA.BA.4075.01A   2950               1            5510         
TCGA.T3.A92M.01A   1679               0            2430         
⋮                ⋮               ⋮               ⋮              
TCGA.CV.7430.01A    3632               0            2857        
TCGA.D6.A74Q.01A   2368               0            3792         
TCGA.CV.7255.01A    5443               0            3542        
TCGA.CQ.6223.01A    1406               1            1875        
TCGA.CV.A460.01A   1049               0            2070         
TCGA.CV.7261.01A   2048               2            1007         
TCGA.CN.4736.01A    980              35            2072         
TCGA.CR.7404.01A   2646              18            2293         
TCGA.UF.A719.01A   5193               0            1751         
TCGA.CV.7236.01A   2366               0            1862         
TCGA.CV.6951.01A   3280               1            1973         
TCGA.CN.4728.01A    1136               1            1739        
TCGA.CV.7090.01A    1050               3            1520        
TCGA.CV.5979.01A    1416               0            1383        
TCGA.CQ.7069.01A    728               0            1603         
TCGA.CN.4733.01A     727               1            1151        
TCGA.CR.7390.01A   1995               1            1561         
TCGA.CV.A45X.01A    1795               1            1913        
TCGA.IQ.A6SH.01A    1428               0            2404        
TCGA.CV.6962.11A   11809               4            1623        
TCGA.CV.6960.11A    3340               0            1388        
TCGA.CV.A464.01A     929               2            2332        
TCGA.C9.A47Z.01A    1707               0            1926        
TCGA.CN.6010.01A   2218               0            1574         
TCGA.WA.A7GZ.11A    2537              41            1171        
TCGA.CV.7235.01A    492               0             976         
TCGA.CX.7086.01A    2741               0            1952        
TCGA.CV.6935.11A    8492               3            1578        
TCGA.P3.A6SW.01A    770               1            1569         
TCGA.HD.A6HZ.01A    923               0            1

In [9]:
df_Puram_t <- t(df_Puram)
colnames(df_Puram_t)<-df_Puram_t[1,]
df_Puram_t <- df_Puram_t[-1,]

In [10]:
df_Puram_t

ENSG00000121410 ENSG00000268895 ENSG00000148584
HN28_P15_D06_S330_comb     0.0000000       0.0000000       0.0000000     
HN28_P6_G05_S173_comb      0.0000000       0.0000000       0.0000000     
HN26_P14_D11_S239_comb     0.0000000       0.0000000       0.0000000     
HN26_P14_H05_S281_comb     0.0000000       0.0000000       0.0000000     
HN26_P25_H09_S189_comb     0.0000000       0.0000000       0.0000000     
HN26_P14_H06_S282_comb     0.0662610       0.0000000       0.0000000     
HN25_P25_C04_S316_comb     0.0000000       0.0000000       0.0000000     
HN26_P25_A11_S107_comb     0.5410200       0.0000000       0.0000000     
HN26_P25_C09_S129_comb     0.0000000       0.0000000       0.0000000     
HNSCC26_P24_H05_S377_comb  1.3004000       0.0000000       0.0000000     
HN25_P5_E10_S58_comb       1.983300        0.000000        0.000000      
HN25_P24_B02_S110_comb     0.0000000       0.0000000       0.0000000     
HN25_P25_C01_S313_comb     0.0000000       0.0000000       0.0000000     
HN28_P15_G06_S366_comb     0.0000000       0.0000000       0.0000000     
HN26_P6_B06_S18_comb       0.0000000       0.0000000       0.0000000     
HN25_P5_C05_S29_comb       0.0000000       0.0000000       0.0000000     
HN28_P6_D09_S141_comb      0.000000        0.000000        0.000000      
HN26_P5_H01_S85_comb       0.0000000       0.0000000       0.0000000     
HN26_P13_D07_S139_comb     0.0000000       0.0000000       0.0000000     
HN26_P5_E06_S54_comb       0.0000000       0.0000000       0.0000000     
HN26_P26_H05_S377_comb     0.0000000       0.0000000       0.0000000     
HNSCC26_P24_F12_S360_comb  0.0000000       0.0000000       0.0000000     
HNSCC26_P24_F03_S351_comb  0.0000000       0.0000000       0.0000000     
HN26_P6_D08_S44_comb       0.0000000       0.0000000       0.0000000     
HN25_P5_E12_S60_comb       0.0000000       2.0635000       0.0000000     
HNSCC26_P24_C01_S313_comb  0.0000000       0.0000000       0.0000000     
HN25_P5_F01_S61_comb       0.0000000       0.0000000       0.0000000     
HN26_P5_D12_S48_comb       0.000000        0.000000        0.000000      
HN25_P25_C11_S323_comb     0.0000000       0.0000000       0.0214800     
HN25_P25_B12_S312_comb     0.0000000       0.0000000       0.0000000     
⋮                         ⋮               ⋮               ⋮              
HNSCC17_P4_A08_S104_comb   0.0000000       0.0000000       0.0000000     
HNSCC17_P4_E10_S154_comb   0.0000000       0.0000000       0.0000000     
HNSCC17_P4_E02_S146_comb   0.000000        0.000000        0.000000      
HNSCC20_P13_F10_S262_comb  0.0000000       0.0000000       0.0000000     
HNSCC17_P4_D04_S136_comb   0.0000000       0.0000000       0.0000000     
HNSCC20_P13_C06_S222_comb  0.0000000       0.0000000       0.0468400     
HNSCC20_P13_C12_S228_comb  1.112400        0.000000        0.000000      
HNSCC17_P4_H04_S184_comb   0.0000000       0.0000000       0.0000000     
HNSCC17_P4_C06_S126_comb   0.9913200       0.0000000       0.0000000     
HNSCC20_P3_D01_S37_comb    0.0000000       0.0000000       0.0000000     
HNSCC20_P3_D09_S45_comb    0.000000        0.000000        0.000000      
HNSCC17_P4_E06_S150_comb   0.0000000       0.0000000       0.0000000     
HNSCC20_P13_H10_S286_comb  0.0000000       0.0000000       0.0482360     
HNSCC17_P4_A02_S98_comb    0.0000000       0.0000000       0.0000000     
HNSCC20_P13_D01_S229_comb  0.0000000       0.0000000       0.0000000     
HNSCC20_P13_G02_S266_comb  0.0000000       0.0000000       0.0000000     
HNSCC17_P4_B04_S112_comb   0.0000000       0.0000000       0.0000000     
HNSCC17_P4_C09_S129_comb   0.0000000       0.0000000       0.0000000     
HNSCC20_P13_A12_S204_comb  0.0000000       0.0000000       0.0000000     
HNSCC17_P4_F06_S162_comb   0.0000000       0.0000000       0.0000000     
HNSCC20_P3_B10_S22_comb    0.6332000       0.0000000       0.0000000     
HNSCC20_P13_B11_S215_comb  1.2259000       1.0642000       0.0000000     
HNSCC20_P3_C08_S32_comb    1.3

In [49]:
label <- read.table("./Puram/PuramHNSCC_cell_category_mappedCelltypes.csv",sep=',',header=TRUE)

In [12]:
# celltypelab <- row.names(df_Puram_t)
# head(celltypelab)

In [50]:
label

X                         cell_category
1    HN28_P15_D06_S330_comb    Fibroblast   
2    HN28_P6_G05_S173_comb     Fibroblast   
3    HN26_P14_D11_S239_comb    tumor        
4    HN26_P14_H05_S281_comb    Fibroblast   
5    HN26_P25_H09_S189_comb    tumor        
6    HN26_P14_H06_S282_comb    tumor        
7    HN25_P25_C04_S316_comb    tumor        
8    HN26_P25_A11_S107_comb    Fibroblast   
9    HN26_P25_C09_S129_comb    tumor        
10   HNSCC26_P24_H05_S377_comb tumor        
11   HN25_P5_E10_S58_comb      Fibroblast   
12   HN25_P24_B02_S110_comb    tumor        
13   HN25_P25_C01_S313_comb    B cell       
14   HN28_P15_G06_S366_comb    Fibroblast   
15   HN26_P6_B06_S18_comb      Fibroblast   
16   HN25_P5_C05_S29_comb      Fibroblast   
17   HN28_P6_D09_S141_comb     Fibroblast   
18   HN26_P5_H01_S85_comb      tumor        
19   HN26_P13_D07_S139_comb    Fibroblast   
20   HN26_P5_E06_S54_comb      tumor        
21   HN26_P26_H05_S377_comb    tumor        
22   HNSCC26_P24_F12_S360_comb tumor        
23   HNSCC26_P24_F03_S351_comb tumor        
24   HN26_P6_D08_S44_comb      tumor        
25   HN25_P5_E12_S60_comb      Fibroblast   
26   HNSCC26_P24_C01_S313_comb tumor        
27   HN25_P5_F01_S61_comb      Fibroblast   
28   HN26_P5_D12_S48_comb      Fibroblast   
29   HN25_P25_C11_S323_comb    Fibroblast   
30   HN25_P25_B12_S312_comb    tumor        
⋮    ⋮                         ⋮            
5873 HNSCC17_P4_A08_S104_comb  tumor        
5874 HNSCC17_P4_E10_S154_comb  tumor        
5875 HNSCC17_P4_E02_S146_comb  tumor        
5876 HNSCC20_P13_F10_S262_comb tumor        
5877 HNSCC17_P4_D04_S136_comb  tumor        
5878 HNSCC20_P13_C06_S222_comb tumor        
5879 HNSCC20_P13_C12_S228_comb tumor        
5880 HNSCC17_P4_H04_S184_comb  tumor        
5881 HNSCC17_P4_C06_S126_comb  tumor        
5882 HNSCC20_P3_D01_S37_comb   tumor        
5883 HNSCC20_P3_D09_S45_comb   tumor        
5884 HNSCC17_P4_E06_S150_comb  tumor        
5885 HNSCC20_P13_H10_S286_comb tumor        
5886 HNSCC17_P4_A02_S98_comb   NA           
5887 HNSCC20_P13_D01_S229_comb tumor        
5888 HNSCC20_P13_G02_S266_comb tumor        
5889 HNSCC17_P4_B04_S112_comb  tumor        
5890 HNSCC17_P4_C09_S129_comb  tumor        
5891 HNSCC20_P13_A12_S204_comb tumor        
5892 HNSCC17_P4_F06_S162_comb  tumor        
5893 HNSCC20_P3_B10_S22_comb   tumor        
5894 HNSCC20_P13_B11_S215_comb tumor        
5895 HNSCC20_P3_C08_S32_comb   tumor        
5896 HNSCC17_P4_H03_S183_comb  tumor        
5897 HNSCC20_P3_F09_S69_comb   tumor        
5898 HNSCC17_P4_G12_S180_comb  NA           
5899 HNSCC20_P13_C05_S221_comb tumor        
5900 HNSCC17_P4_C12_S132_comb  tumor        
5901 HNSCC20_P3_H08_S92_comb   tumor        
5902 HNSCC20_P3_G06_S78_comb   tumor

In [13]:
# fixing the problem summing characters in df
class(df_Puram_t)<-'numeric'

In [14]:
class(df_TCGA_t)<-'numeric'

In [15]:
# class(df_Puram_t)

[1] "matrix" "array"

#### Run cleanup procedure, to remove "RB"=ribosomal genes; "chrM"=mitochondrial genes; "chrX" and "chrY" denoting genes on the sex chromosome.
- 18332 genes left, 
- 5902 single cells, 547 bulk samples
- 9 celltypes (all tumors are same), 1 NA type

In [17]:
ref.data.filtered <- cleanup.genes(ref.dat=df_Puram_t, gene.type=c("RB","chrM","chrX","chrY"), species="hs", input.type="scRNA",exp.cells=5)

[1] "EMSEMBLE IDs detected. Cleaning up genes based on EMSEMBLE IDs."
A total of  1226  genes from RB chrM chrX chrY  have been excluded 
A total of  2255  lowly expressed genes have been excluded 


In [37]:
ref.data.filtered

ENSG00000121410 ENSG00000268895 ENSG00000148584
HN28_P15_D06_S330_comb    0.000000        0.0000          0.00000        
HN28_P6_G05_S173_comb     0.000000        0.0000          0.00000        
HN26_P14_D11_S239_comb    0.000000        0.0000          0.00000        
HN26_P14_H05_S281_comb    0.000000        0.0000          0.00000        
HN26_P25_H09_S189_comb    0.000000        0.0000          0.00000        
HN26_P14_H06_S282_comb    0.066261        0.0000          0.00000        
HN25_P25_C04_S316_comb    0.000000        0.0000          0.00000        
HN26_P25_A11_S107_comb    0.541020        0.0000          0.00000        
HN26_P25_C09_S129_comb    0.000000        0.0000          0.00000        
HNSCC26_P24_H05_S377_comb 1.300400        0.0000          0.00000        
HN25_P5_E10_S58_comb      1.983300        0.0000          0.00000        
HN25_P24_B02_S110_comb    0.000000        0.0000          0.00000        
HN25_P25_C01_S313_comb    0.000000        0.0000          0.00000        
HN28_P15_G06_S366_comb    0.000000        0.0000          0.00000        
HN26_P6_B06_S18_comb      0.000000        0.0000          0.00000        
HN25_P5_C05_S29_comb      0.000000        0.0000          0.00000        
HN28_P6_D09_S141_comb     0.000000        0.0000          0.00000        
HN26_P5_H01_S85_comb      0.000000        0.0000          0.00000        
HN26_P13_D07_S139_comb    0.000000        0.0000          0.00000        
HN26_P5_E06_S54_comb      0.000000        0.0000          0.00000        
HN26_P26_H05_S377_comb    0.000000        0.0000          0.00000        
HNSCC26_P24_F12_S360_comb 0.000000        0.0000          0.00000        
HNSCC26_P24_F03_S351_comb 0.000000        0.0000          0.00000        
HN26_P6_D08_S44_comb      0.000000        0.0000          0.00000        
HN25_P5_E12_S60_comb      0.000000        2.0635          0.00000        
HNSCC26_P24_C01_S313_comb 0.000000        0.0000          0.00000        
HN25_P5_F01_S61_comb      0.000000        0.0000          0.00000        
HN26_P5_D12_S48_comb      0.000000        0.0000          0.00000        
HN25_P25_C11_S323_comb    0.000000        0.0000          0.02148        
HN25_P25_B12_S312_comb    0.000000        0.0000          0.00000        
⋮                         ⋮               ⋮               ⋮              
HNSCC17_P4_A08_S104_comb  0.00000         0.0000          0.000000       
HNSCC17_P4_E10_S154_comb  0.00000         0.0000          0.000000       
HNSCC17_P4_E02_S146_comb  0.00000         0.0000          0.000000       
HNSCC20_P13_F10_S262_comb 0.00000         0.0000          0.000000       
HNSCC17_P4_D04_S136_comb  0.00000         0.0000          0.000000       
HNSCC20_P13_C06_S222_comb 0.00000         0.0000          0.046840       
HNSCC20_P13_C12_S228_comb 1.11240         0.0000          0.000000       
HNSCC17_P4_H04_S184_comb  0.00000         0.0000          0.000000       
HNSCC17_P4_C06_S126_comb  0.99132         0.0000          0.000000       
HNSCC20_P3_D01_S37_comb   0.00000         0.0000          0.000000       
HNSCC20_P3_D09_S45_comb   0.00000         0.0000          0.000000       
HNSCC17_P4_E06_S150_comb  0.00000         0.0000          0.000000       
HNSCC20_P13_H10_S286_comb 0.00000         0.0000          0.048236       
HNSCC17_P4_A02_S98_comb   0.00000         0.0000          0.000000       
HNSCC20_P13_D01_S229_comb 0.00000         0.0000          0.000000       
HNSCC20_P13_G02_S266_comb 0.00000         0.0000          0.000000       
HNSCC17_P4_B04_S112_comb  0.00000         0.0000          0.000000       
HNSCC17_P4_C09_S129_comb  0.00000         0.0000          0.000000       
HNSCC20_P13_A12_S204_comb 0.00000         0.0000          0.000000       
HNSCC17_P4_F06_S162_comb  0.00000         0.0000          0.000000       
HNSCC20_P3_B10_S22_comb   0.63320         0.0000          0.000000       
HNSCC20_P13_B11_S215_comb 1.22590         1.0642          0.000000       
HNSCC20_P3_C08_S32_comb   1.37

In [51]:
label

X                         cell_category
1    HN28_P15_D06_S330_comb    Fibroblast   
2    HN28_P6_G05_S173_comb     Fibroblast   
3    HN26_P14_D11_S239_comb    tumor        
4    HN26_P14_H05_S281_comb    Fibroblast   
5    HN26_P25_H09_S189_comb    tumor        
6    HN26_P14_H06_S282_comb    tumor        
7    HN25_P25_C04_S316_comb    tumor        
8    HN26_P25_A11_S107_comb    Fibroblast   
9    HN26_P25_C09_S129_comb    tumor        
10   HNSCC26_P24_H05_S377_comb tumor        
11   HN25_P5_E10_S58_comb      Fibroblast   
12   HN25_P24_B02_S110_comb    tumor        
13   HN25_P25_C01_S313_comb    B cell       
14   HN28_P15_G06_S366_comb    Fibroblast   
15   HN26_P6_B06_S18_comb      Fibroblast   
16   HN25_P5_C05_S29_comb      Fibroblast   
17   HN28_P6_D09_S141_comb     Fibroblast   
18   HN26_P5_H01_S85_comb      tumor        
19   HN26_P13_D07_S139_comb    Fibroblast   
20   HN26_P5_E06_S54_comb      tumor        
21   HN26_P26_H05_S377_comb    tumor        
22   HNSCC26_P24_F12_S360_comb tumor        
23   HNSCC26_P24_F03_S351_comb tumor        
24   HN26_P6_D08_S44_comb      tumor        
25   HN25_P5_E12_S60_comb      Fibroblast   
26   HNSCC26_P24_C01_S313_comb tumor        
27   HN25_P5_F01_S61_comb      Fibroblast   
28   HN26_P5_D12_S48_comb      Fibroblast   
29   HN25_P25_C11_S323_comb    Fibroblast   
30   HN25_P25_B12_S312_comb    tumor        
⋮    ⋮                         ⋮            
5873 HNSCC17_P4_A08_S104_comb  tumor        
5874 HNSCC17_P4_E10_S154_comb  tumor        
5875 HNSCC17_P4_E02_S146_comb  tumor        
5876 HNSCC20_P13_F10_S262_comb tumor        
5877 HNSCC17_P4_D04_S136_comb  tumor        
5878 HNSCC20_P13_C06_S222_comb tumor        
5879 HNSCC20_P13_C12_S228_comb tumor        
5880 HNSCC17_P4_H04_S184_comb  tumor        
5881 HNSCC17_P4_C06_S126_comb  tumor        
5882 HNSCC20_P3_D01_S37_comb   tumor        
5883 HNSCC20_P3_D09_S45_comb   tumor        
5884 HNSCC17_P4_E06_S150_comb  tumor        
5885 HNSCC20_P13_H10_S286_comb tumor        
5886 HNSCC17_P4_A02_S98_comb   NA           
5887 HNSCC20_P13_D01_S229_comb tumor        
5888 HNSCC20_P13_G02_S266_comb tumor        
5889 HNSCC17_P4_B04_S112_comb  tumor        
5890 HNSCC17_P4_C09_S129_comb  tumor        
5891 HNSCC20_P13_A12_S204_comb tumor        
5892 HNSCC17_P4_F06_S162_comb  tumor        
5893 HNSCC20_P3_B10_S22_comb   tumor        
5894 HNSCC20_P13_B11_S215_comb tumor        
5895 HNSCC20_P3_C08_S32_comb   tumor        
5896 HNSCC17_P4_H03_S183_comb  tumor        
5897 HNSCC20_P3_F09_S69_comb   tumor        
5898 HNSCC17_P4_G12_S180_comb  NA           
5899 HNSCC20_P13_C05_S221_comb tumor        
5900 HNSCC17_P4_C12_S132_comb  tumor        
5901 HNSCC20_P3_H08_S92_comb   tumor        
5902 HNSCC20_P3_G06_S78_comb   tumor

In [127]:
# ?run.Ted

In [52]:
#run BayesPrism main func
res <- run.Ted(ref.dat=ref.data.filtered, X=df_TCGA_t,cell.type.labels=label$cell_category,
               tum.key='tumor',input.type="scRNA",n.cores=12,pdf.name="PuramTCGA",seed=52322)

[1] "removing non-numeric genes..."
[1] "removing outlier genes..."
Number of outlier genes filtered= 7 
[1] "aligning reference and mixture..."
[1] "Warning: NA found in the cell (sub)type labels. These cells will be excluded!"
[1] "Warning: NA found in the cell (sub)type labels. These cells will be excluded!"
[1] "run first sampling"
Start run... This may take a while 
R Version:  R version 4.1.3 (2022-03-10) 



snowfall 1.84-6.1 initialized (using snow 0.4-4): parallel execution on 12 CPUs.



Stopping cluster




[1] "merge subtypes"
        Fibroblast tumor B cell myocyte Macrophage Endothelial T cell Dendritic
Min.         0.000 0.003  0.000   0.000      0.000       0.000  0.000     0.000
1st Qu.      0.047 0.618  0.000   0.000      0.007       0.009  0.000     0.000
Median       0.129 0.767  0.000   0.000      0.017       0.021  0.001     0.000
Mean         0.164 0.732  0.002   0.026      0.032       0.032  0.006     0.003
3rd Qu.      0.240 0.882  0.000   0.012      0.045       0.041  0.005     0.000
Max.         0.958 0.998  0.053   0.785      0.287       0.406  0.126     0.090
         Mast
Min.    0.000
1st Qu. 0.000
Median  0.000
Mean    0.004
3rd Qu. 0.002
Max.    0.129
[1] "pooling information across samples"
Start optimization... This may take a while 


snowfall 1.84-6.1 initialized (using snow 0.4-4): parallel execution on 12 CPUs.



Stopping cluster




[1] "run final sampling"
Start run... This may take a while 


snowfall 1.84-6.1 initialized (using snow 0.4-4): parallel execution on 12 CPUs.



Stopping cluster




        tumor Fibroblast B cell myocyte Macrophage Endothelial T cell Dendritic
Min.    0.072      0.000  0.000   0.000      0.000       0.000  0.000     0.000
1st Qu. 0.726      0.022  0.000   0.000      0.005       0.003  0.000     0.000
Median  0.851      0.074  0.000   0.000      0.013       0.009  0.000     0.000
Mean    0.805      0.121  0.002   0.021      0.025       0.016  0.004     0.004
3rd Qu. 0.928      0.169  0.001   0.006      0.034       0.021  0.004     0.000
Max.    0.999      0.862  0.046   0.799      0.223       0.186  0.075     0.111
         Mast
Min.    0.000
1st Qu. 0.000
Median  0.000
Mean    0.002
3rd Qu. 0.000
Max.    0.099
[1] "vst transformation is feasible"


converting counts to integer mode



In [54]:
saveRDS(res,"/home/cke/BayesPrism/BP_PuramTCGA_res.rds")

### A list of result is returned in this function.

para	
All input data and parameters.

res	
All output of TED.

res$first.gibbs.res$gibbs.theta	
Initial estimates of fraction for all cell subtypes in each bulk sample.

res$first.gibbs.res$Znkg	
Initial estimates of the mean of posterior read count for each cell subtypes in each bulk sample.

res$first.gibbs.res$theta.merged	
Initial estimates of fraction summed across cell types in each bulk sample.

res$first.gibbs.res$Znkg.merged	
Initial estimates of the mean of posterior read count summed across cell types in each bulk sample.

res$Zkg.tum	
Mean of posterior of gene expression of tumor in each patient.

res$Zkg.tum.norm	
Depth normalized Zkg.tum (A pseudo count is added, such that the zero-valued genes have the same value as the min(phi.input)). Refered to as the psi.tum in the TED paper)

res$Zkg.tum.vst	
Variance stablized transformed value of Zkg.tum. If vst transformation is not feasible, return NULL.

res$phi.env	
Batch effect corrected expression profiles of stromal cells (refered to as the psi.str in the TED paper)

res$final.gibbs.theta	
Updated theta after batch correction and tumor expression estimates.

res$cor.mat	
The correlation matrix of the estimated tumor expression profiles across bulk RNA-seq samples.

In [61]:
#final est cell type fraction in each bulk samples
res$res$final.gibbs.thet

tumor     Fibroblast   B cell       myocyte      Macrophage  
TCGA.BB.4224.01A 0.8953963 3.295279e-03 1.535432e-07 2.076378e-03 2.663471e-06
TCGA.H7.7774.01A 0.9807876 9.213523e-07 1.956431e-06 1.582816e-07 1.186069e-02
TCGA.CV.6943.01A 0.5239461 3.799039e-01 1.548700e-03 2.208963e-07 7.545321e-02
TCGA.CN.5374.01A 0.8727039 2.463040e-02 6.619988e-03 6.413547e-03 2.021994e-02
TCGA.CQ.6227.01A 0.6538779 2.342006e-01 1.996756e-07 7.141557e-02 7.623701e-03
TCGA.CV.6959.01A 0.7848157 9.008879e-02 3.630708e-05 6.450137e-02 2.570535e-02
TCGA.F7.A61V.01A 0.9545372 1.147164e-02 1.391773e-03 2.241424e-02 4.397206e-03
TCGA.CV.7413.01A 0.7576784 2.165254e-01 1.947499e-07 8.478491e-08 2.579297e-02
TCGA.CV.7247.01A 0.7911627 1.108126e-01 1.374201e-07 7.858550e-02 6.057291e-07
TCGA.CR.5249.01A 0.8287251 5.001096e-02 7.550117e-03 1.137713e-07 2.663733e-02
TCGA.CQ.5331.01A 0.8679074 3.680547e-02 6.729025e-04 6.492139e-08 6.722354e-02
TCGA.BB.8601.01A 0.9933812 4.261229e-03 7.357805e-05 2.507918e-07 3.730104e-06
TCGA.CV.A45W.01A 0.7583385 9.038100e-02 4.573747e-02 2.842859e-07 5.930226e-02
TCGA.CQ.7063.01A 0.8932472 2.143019e-02 2.929627e-07 2.839658e-07 6.462695e-02
TCGA.CN.5373.01A 0.7648254 1.857341e-01 3.306371e-07 6.088100e-04 2.179955e-02
TCGA.CV.A45T.01A 0.8189485 5.773553e-02 5.785000e-03 1.473200e-02 6.505784e-02
TCGA.HD.A4C1.01A 0.9037093 8.132439e-02 7.313362e-07 1.059151e-07 4.587139e-03
TCGA.CV.7438.11A 0.9736846 2.405088e-03 6.825740e-04 2.766721e-03 1.248155e-02
TCGA.CV.5444.01A 0.4018990 5.008603e-01 1.421520e-07 3.634083e-02 5.472763e-02
TCGA.BA.5556.01A 0.8490606 4.796692e-02 2.149780e-05 9.793571e-08 5.747951e-02
TCGA.P3.A6SX.01A 0.8960362 9.110862e-02 1.878011e-07 1.098512e-02 6.988049e-07
TCGA.CV.7242.11A 0.9755767 5.311162e-03 3.153993e-05 3.141680e-07 1.229840e-02
TCGA.DQ.7592.01A 0.8323528 1.342008e-01 1.722611e-07 1.280968e-02 1.281682e-02
TCGA.CV.6937.01A 0.8360634 1.441721e-01 3.585737e-07 4.275409e-07 1.778293e-02
TCGA.QK.A8ZB.01A 0.8499897 1.257516e-01 6.130647e-03 1.031490e-07 6.951394e-03
TCGA.CV.7432.01A 0.9688731 2.165490e-02 1.442955e-07 1.739435e-07 4.331192e-03
TCGA.CV.7089.01A 0.9668070 8.355609e-03 2.990496e-07 1.403883e-07 8.618054e-03
TCGA.CV.6939.11A 0.3108183 2.925750e-02 1.597606e-03 6.230553e-01 1.509185e-03
TCGA.BA.4075.01A 0.9198085 4.745561e-02 3.472322e-07 2.176400e-03 2.556290e-02
TCGA.T3.A92M.01A 0.6663292 2.896026e-01 1.676743e-03 1.417736e-07 3.400842e-02
⋮                ⋮         ⋮            ⋮            ⋮            ⋮           
TCGA.CV.7430.01A 0.9774183 0.015130849  3.765206e-07 1.369697e-07 6.725784e-04
TCGA.D6.A74Q.01A 0.8842049 0.085482558  6.163754e-03 1.556808e-07 7.739195e-03
TCGA.CV.7255.01A 0.9440443 0.027434705  2.563307e-07 9.352016e-03 1.086210e-06
TCGA.CQ.6223.01A 0.9781348 0.006737177  3.301923e-04 1.067407e-07 1.298400e-02
TCGA.CV.A460.01A 0.7211691 0.150885458  2.310984e-03 1.378379e-07 3.647569e-02
TCGA.CV.7261.01A 0.7686271 0.199999998  1.800759e-03 1.830788e-07 1.143315e-02
TCGA.CN.4736.01A 0.5770123 0.202003425  3.214170e-07 1.087496e-01 8.823329e-02
TCGA.CR.7404.01A 0.8335180 0.049152821  5.844858e-03 1.735372e-07 4.436907e-02
TCGA.UF.A719.01A 0.9767740 0.009284403  1.501744e-05 6.696999e-07 5.670974e-03
TCGA.CV.7236.01A 0.7583639 0.201827735  3.707119e-07 7.813978e-08 1.085050e-02
TCGA.CV.6951.01A 0.9146561 0.059564714  4.368603e-07 2.203449e-02 1.716345e-03
TCGA.CN.4728.01A 0.2860686 0.608054579  2.411349e-07 5.917761e-03 9.206015e-02
TCGA.CV.7090.01A 0.7096929 0.227983815  1.494937e-07 1.164804e-07 5.363601e-02
TCGA.CV.5979.01A 0.9170363 0.069411556  2.913286e-07 1.650445e-03 4.982065e-03
TCGA.CQ.7069.01A 0.8867397 0.091163538  3.126233e-07 1.693463e-07 8.315407e-04
TCGA.CN.4733.01A 0.4590225 0.313208243  2.204239e-02 1.200962e-02 1.121640e-01
TCGA.CR.7390.01A 0.8226924 0.170653482  1.782093e-07 1.786800e-07 3.298995e-03
TCGA.CV.A45X.01A 0.9860628 0.005683827  8.349780e-07 3.304159e-07 7.281100e-03
TCGA.IQ.A6SH.01A 0.9423716 0.028385095  1.185881e-06 2.116455e-07 2

In [65]:
res$para

In [67]:
res1 <- readRDS("/home/cke/BayesPrism/BP_PuramTCGA_res.rds")

In [68]:
res1$res$final.gibbs.thet

tumor     Fibroblast   B cell       myocyte      Macrophage  
TCGA.BB.4224.01A 0.8953963 3.295279e-03 1.535432e-07 2.076378e-03 2.663471e-06
TCGA.H7.7774.01A 0.9807876 9.213523e-07 1.956431e-06 1.582816e-07 1.186069e-02
TCGA.CV.6943.01A 0.5239461 3.799039e-01 1.548700e-03 2.208963e-07 7.545321e-02
TCGA.CN.5374.01A 0.8727039 2.463040e-02 6.619988e-03 6.413547e-03 2.021994e-02
TCGA.CQ.6227.01A 0.6538779 2.342006e-01 1.996756e-07 7.141557e-02 7.623701e-03
TCGA.CV.6959.01A 0.7848157 9.008879e-02 3.630708e-05 6.450137e-02 2.570535e-02
TCGA.F7.A61V.01A 0.9545372 1.147164e-02 1.391773e-03 2.241424e-02 4.397206e-03
TCGA.CV.7413.01A 0.7576784 2.165254e-01 1.947499e-07 8.478491e-08 2.579297e-02
TCGA.CV.7247.01A 0.7911627 1.108126e-01 1.374201e-07 7.858550e-02 6.057291e-07
TCGA.CR.5249.01A 0.8287251 5.001096e-02 7.550117e-03 1.137713e-07 2.663733e-02
TCGA.CQ.5331.01A 0.8679074 3.680547e-02 6.729025e-04 6.492139e-08 6.722354e-02
TCGA.BB.8601.01A 0.9933812 4.261229e-03 7.357805e-05 2.507918e-07 3.730104e-06
TCGA.CV.A45W.01A 0.7583385 9.038100e-02 4.573747e-02 2.842859e-07 5.930226e-02
TCGA.CQ.7063.01A 0.8932472 2.143019e-02 2.929627e-07 2.839658e-07 6.462695e-02
TCGA.CN.5373.01A 0.7648254 1.857341e-01 3.306371e-07 6.088100e-04 2.179955e-02
TCGA.CV.A45T.01A 0.8189485 5.773553e-02 5.785000e-03 1.473200e-02 6.505784e-02
TCGA.HD.A4C1.01A 0.9037093 8.132439e-02 7.313362e-07 1.059151e-07 4.587139e-03
TCGA.CV.7438.11A 0.9736846 2.405088e-03 6.825740e-04 2.766721e-03 1.248155e-02
TCGA.CV.5444.01A 0.4018990 5.008603e-01 1.421520e-07 3.634083e-02 5.472763e-02
TCGA.BA.5556.01A 0.8490606 4.796692e-02 2.149780e-05 9.793571e-08 5.747951e-02
TCGA.P3.A6SX.01A 0.8960362 9.110862e-02 1.878011e-07 1.098512e-02 6.988049e-07
TCGA.CV.7242.11A 0.9755767 5.311162e-03 3.153993e-05 3.141680e-07 1.229840e-02
TCGA.DQ.7592.01A 0.8323528 1.342008e-01 1.722611e-07 1.280968e-02 1.281682e-02
TCGA.CV.6937.01A 0.8360634 1.441721e-01 3.585737e-07 4.275409e-07 1.778293e-02
TCGA.QK.A8ZB.01A 0.8499897 1.257516e-01 6.130647e-03 1.031490e-07 6.951394e-03
TCGA.CV.7432.01A 0.9688731 2.165490e-02 1.442955e-07 1.739435e-07 4.331192e-03
TCGA.CV.7089.01A 0.9668070 8.355609e-03 2.990496e-07 1.403883e-07 8.618054e-03
TCGA.CV.6939.11A 0.3108183 2.925750e-02 1.597606e-03 6.230553e-01 1.509185e-03
TCGA.BA.4075.01A 0.9198085 4.745561e-02 3.472322e-07 2.176400e-03 2.556290e-02
TCGA.T3.A92M.01A 0.6663292 2.896026e-01 1.676743e-03 1.417736e-07 3.400842e-02
⋮                ⋮         ⋮            ⋮            ⋮            ⋮           
TCGA.CV.7430.01A 0.9774183 0.015130849  3.765206e-07 1.369697e-07 6.725784e-04
TCGA.D6.A74Q.01A 0.8842049 0.085482558  6.163754e-03 1.556808e-07 7.739195e-03
TCGA.CV.7255.01A 0.9440443 0.027434705  2.563307e-07 9.352016e-03 1.086210e-06
TCGA.CQ.6223.01A 0.9781348 0.006737177  3.301923e-04 1.067407e-07 1.298400e-02
TCGA.CV.A460.01A 0.7211691 0.150885458  2.310984e-03 1.378379e-07 3.647569e-02
TCGA.CV.7261.01A 0.7686271 0.199999998  1.800759e-03 1.830788e-07 1.143315e-02
TCGA.CN.4736.01A 0.5770123 0.202003425  3.214170e-07 1.087496e-01 8.823329e-02
TCGA.CR.7404.01A 0.8335180 0.049152821  5.844858e-03 1.735372e-07 4.436907e-02
TCGA.UF.A719.01A 0.9767740 0.009284403  1.501744e-05 6.696999e-07 5.670974e-03
TCGA.CV.7236.01A 0.7583639 0.201827735  3.707119e-07 7.813978e-08 1.085050e-02
TCGA.CV.6951.01A 0.9146561 0.059564714  4.368603e-07 2.203449e-02 1.716345e-03
TCGA.CN.4728.01A 0.2860686 0.608054579  2.411349e-07 5.917761e-03 9.206015e-02
TCGA.CV.7090.01A 0.7096929 0.227983815  1.494937e-07 1.164804e-07 5.363601e-02
TCGA.CV.5979.01A 0.9170363 0.069411556  2.913286e-07 1.650445e-03 4.982065e-03
TCGA.CQ.7069.01A 0.8867397 0.091163538  3.126233e-07 1.693463e-07 8.315407e-04
TCGA.CN.4733.01A 0.4590225 0.313208243  2.204239e-02 1.200962e-02 1.121640e-01
TCGA.CR.7390.01A 0.8226924 0.170653482  1.782093e-07 1.786800e-07 3.298995e-03
TCGA.CV.A45X.01A 0.9860628 0.005683827  8.349780e-07 3.304159e-07 7.281100e-03
TCGA.IQ.A6SH.01A 0.9423716 0.028385095  1.185881e-06 2.116455e-07 2